packages/job-worker/src/playout/abPlayback/index.ts

```diff
@@ -103,3 +122,36 @@ export function applyAbPlaybackForTimeline(
 
 	return newAbSessionsResult
 }
+
+function findPlayersInRouteSets(routeSets: Record<string, StudioRouteSet>): MembersOfRouteSets[] {
+	const players: MembersOfRouteSets[] = []
+	for (const [_key, routeSet] of Object.entries<StudioRouteSet>(routeSets)) {
+		routeSet.abPlayers.forEach((abPlayer) => {
+			players.push({
+				playerId: abPlayer.playerId,
+				poolName: abPlayer.poolName,
+				disabled: !routeSet.active,
+			})
+		})
+	}
+	return players
+}
+
+function abPoolFilterDisabled(
+	context: JobContext,
+	poolName: string,
+	players: ABPlayerDefinition[],
+	membersOfRouteSets: MembersOfRouteSets[]
+): ABPlayerDefinition[] {
+	if (membersOfRouteSets.length == 0) return players
+
+	// Filter out any disabled players:
+	return players.filter((player) => {
+		const disabled = membersOfRouteSets.find((abPlayer) => abPlayer.playerId === player.playerId)?.disabled
```

Comment:
what should happen if multiple routesets reference a player?

Reply:
I think this needs a discussion whether it's needed or there should be a check for this in settings, or something else

Reply:
Yeah I think it needs a discussion. 
Currently it will be picking the first value. Which routeset that comes from may not be stable.

I can see this feature is likely to be used with a setup of routesets in an exclusive group of:
* All players (A + B + C + D)
* Server 1 only (A + B)
* Server 2 only (C + D)
in which case, this check won't work reliably. once it gets switched to either 'Server X only', this `find` will fetch the first entry in the array which will be the disabled entry from 'All players' resulting in all 4 players being disabled

But the current implementation would be fine for a set of toggles which each control whether a single player is available.